app/code/Magento/Catalog/Model/Category/Attribute/Source/Layout.php

```diff
@@ -26,14 +32,14 @@ public function __construct(\Magento\Framework\View\Model\PageLayout\Config\Buil
     }
 
     /**
-     * {@inheritdoc}
+     * @inheritdoc
      */
     public function getAllOptions()
     {
-        if (!$this->_options) {
```

Comment:
It looks a bit wired to set a value to the property and avoid using it. I think existing method body is more clear and avoids additional options processing.

Reply:
There is no any additional "option processing".

> It looks a bit wired to set a value to the property and avoid using it

This is what current recommendation says. It would be much better to just remove property as probability of BC break is negligible, but if you insist on keeping it deprecated - let's just keep it prepared for removal.

Reply:
Hi guys, are you expecting something from me? 

Because I made what was the most logical thing from my point of view:
- deprecate the property
- keep assigning a value to the deprecated property 
- remove internal usages of the property to prepare its removal

But I'm not sure to understand exactly what you  want right now... 😄

Reply:
@ajardin yeah, everything is fine from my point of view. Remaining part is to convince Sergey, no code changes required until we agree upon them.